### DISCUSSION TITLE
Select All Objects in a List - Possible?

### DISCUSSION THREAD

**Discussion by ssandino** [firecmsco/firecms#345] [Ideas]
Is there a way to add the option to select all objects in a list?

<img width="360" alt="Screen Shot 2022-07-21 at 00 32 20" src="https://user-images.githubusercontent.com/6095849/180093853-dc40715b-b794-4158-913d-a2e41b745083.png">

**Comment by fgatti675** [comment]
Hi @ssandino it's a bit tricky, since we use pagination, to deal with collections that could have thousands of entries.
It would make sense to enable the select all toggle when pagination is disabled manually.
Or maybe we could create a 'select all in this page'

**Comment by ssandino** [comment]
Thanks @fgatti675 – I think your proposition: `select all on this page` would already be a big step forward when pagination is enabled. 

I guess this is a common issue in list views with pagination. Gmail has solved this quite nicely: 
1: Select All = all on the page
2: Offers new option to select beyond the current page/view

![firecms-imp](https://user-images.githubusercontent.com/6095849/180440863-03892811-3833-47ff-a845-8e95edab3718.jpg)

**Comment by ssandino** [comment]
Another simple way how to handle the `select all` for a list – without confusing the user (_All_ is really all? Or only all that are shown?). The `select all `option reveals a dropdown with the two options (see image). 

![Image 13 01 23 at 11 18 copy](https://user-images.githubusercontent.com/6095849/212297260-1345c281-3066-417f-b7b0-6c3572debcc7.png)
by Mailchimp
